### PR TITLE
Make Dar robust against missing files

### PR DESCRIPTION
**Pull request by michael**
Why: A common source of the error was that StencilaArchive/TextureArchive crashed because of a missing sheet.xml
What: Make sure, that a DAR can still be opened, and a warning is printed.

### CODE REVIEW

**Comment by oliver----** [comment]
This would not lead to a 'dirty' manifest, would it? Do we want this?

**Comment by michael** [comment]
It removes the missing files as they would cause troubles anyways. If we left those manifest entries there, we would still see the tabs for the missing documents, causing them to break when tried to be opened. As a consequence the archive would be 'cleaned up' automatically. Like when you load and save, the missing entry will be gone from the manifest. Think that is ok, but we should communicate to the user that we cleaned up, or normalised a broken archive.

**Comment by oliver----** [comment]
I mean, would the archive be considered 'dirty'? I.e. saving have an effect?

**Comment by michael** [comment]
Oh good point. That would be the consequence yes. 

So the whole scenario would look like this:

1. Open broken archive
2. Tell the user the archive is broken, do you want to open it anyway? (we can add that ui later, and do it automatically for now)
3. If yes we apply the fixes (which should produce changes to the archive)
4. If the user hits the save button the fixed archive gets stored to disk

**Comment by oliver----** [comment]
That would be desirable. And now my question: does the current implementation already have the effect of marking the DAR as dirty? ;)

**Comment by michael** [comment]
I can perform the manifest fixes on the manifest session instead of on the rawmanifest, so we get that dirty state. That be ok for you?

**Comment by oliver----** [comment]
Maybe that is cleaner... then we have ops too.

**Comment by michael** [comment]
Ok will do.

It's actually good to think about this. We'll have similar situations when loading old archive versions, where we need to clean them up to 'upgrade'. Could be the same mechanism.

**Comment by oliver----** [comment]
Yeah, good point.

**Comment by michael** [comment]
@oliver---- pls review

File: src/TextureArchive.js [hunk 1]
```diff
@@ -22,6 +22,9 @@ export default class TextureArchive extends PersistedDocumentArchive {
 
     entries.forEach(entry => {
       let record = rawArchive.resources[entry.path]
+      if (!record) {
```
Comment (oliver----) [inline src/TextureArchive.js]: Maybe leave a comment here, when this happens.

**Comment by oliver----** [comment]
Apart from that, looking good.

**Comment by oliver----** [comment]
Have done it.

**Comment by oliver----** [comment]
@michael BTW, is there a reason to use hierarchical versions like in  '1.0.0-preview.1.8' instead of just increasing the first number?

**Pull request merged**